Lib/test/test_fractions.py

```diff
@@ -666,6 +666,24 @@ def test_slots(self):
         r = F(13, 7)
         self.assertRaises(AttributeError, setattr, r, 'a', 10)
 
+    def test_gcd_compatibility_branch(self):
+        # Regression test for bugs.python.org/issue39350
+
+        class myint(int):
+            def __mul__(self, other):
+                return myint(int(self)*int(other))
+
+            @property
+            def numerator(self):
+                return self
+
+            @property
+            def denominator(self):
+                return myint(1)
+
+        myhalf = F(myint(1), myint(2))
```

Comment:
I would prefer to see GCD being computed and used, so I propose:

```suggestion
        myhalf = F(myint(1 * 3), myint(2 * 3))
```